### PR TITLE
Fix swallowed errors in various volume packages

### PR DESCRIPTION
**Pull request by alrs**
**What this PR does / why we need it**: Fixes swallowed errors in various volume packages.

**Release note**:
```release-note NONE
```

### CODE REVIEW

**Comment by rootfs** [comment]
please rest of the fixes here as @xiangpengzhao did in  #38947, thanks

**Comment by alrs** [comment]
@rootfs @thockin  I've merged all the PRs together, per your requests.

**Comment by alrs** [comment]
@rootfs @thockin Is there anything I can do to get this merged before code freeze?

File: pkg/volume/util/device_util_linux_test.go [hunk 1]
```diff
@@ -119,7 +119,13 @@ func TestFindDeviceForPath(t *testing.T) {
 	if disk != "sde" {
 		t.Fatalf("disk [%s] didn't match expected sde", disk)
 	}
+	if err != nil {
```
Comment (lavalamp) [inline pkg/volume/util/device_util_linux_test.go]: Check the error before the disk?

Comment (alrs) [reply pkg/volume/util/device_util_linux_test.go]: I've made the change, @lavalamp.

**Comment by alrs** [comment]
/approve no-issue

On Sep 12, 2017 2:44 PM, "Kubernetes Submit Queue" <notifications@github.com>
wrote:

[APPROVALNOTIFIER] This PR is *NOT APPROVED*

This pull-request has been approved by: *alrs
<https://github.com/kubernetes/kubernetes/pull/51387#issuecomment-326694949>*,
*gnufied
<https://github.com/kubernetes/kubernetes/pull/51387#issuecomment-328964980>*,
*saad-ali
<https://github.com/kubernetes/kubernetes/pull/51387#issuecomment-328989400>*

*No associated issue*. Update pull-request body to add a reference to an
issue, or get approval with /approve no-issue

The full list of commands accepted by this bot can be found here
<https://github.com/kubernetes/test-infra/blob/master/commands.md>.
Needs approval from an approver in each of these OWNERS Files:

   - pkg/controller/volume/attachdetach/OWNERS
   <https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/attachdetach/OWNERS>
   [saad-ali]
   - pkg/volume/OWNERS
   <https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/OWNERS>
   [gnufied,saad-ali]

You can indicate your approval by writing /approve in a comment
You can cancel your approval by writing /approve cancel in a comment

—

You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/51387#issuecomment-328992908>,
or mute the thread

**Pull request merged**